beeline/__init__.py

```diff
@@ -74,15 +75,55 @@ def __init__(self,
         if debug:
             self._init_logger()
 
+        def IsClassicKey(writekey):
+            return len(writekey) == 32
+
         # allow setting some values from the environment
         if not writekey:
             writekey = os.environ.get('HONEYCOMB_WRITEKEY', '')
+        if not writekey:
+            self.log(
+                'writekey not set! set the writekey if you want to send data to honeycomb')
 
-        if not dataset:
-            dataset = os.environ.get('HONEYCOMB_DATASET', '')
+        if IsClassicKey(writekey):
+            if not dataset:
+                dataset = os.environ.get('HONEYCOMB_DATASET', '')
+            if not dataset:
+                self.log(
+                    'dataset not set! set a value for dataset if you want to send data to honeycomb')
+        else:
+            if dataset:
+                self.log(
+                    'dataset will be ignored in favor of service name'
+                )
 
+        default_service_name = "unknown_service"
+        process_name = os.environ.get('PROCESS_EXECUTABLE_NAME', '')
```

Comment:
[upstream otel-python](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py#L173) only checks for env var of process name, so mirrored that here